stdlib/format.mli

```diff
@@ -1150,17 +1150,43 @@ val formatter_of_symbolic_output_buffer : symbolic_output_buffer -> formatter
 
 (** {1 Convenience formatting functions.} *)
 
+val pp_print_iter :
+  ?pp_sep:(formatter -> unit -> unit) ->
+  (('a -> 'b) -> 'c -> 'd) ->
+  (formatter -> 'a -> 'b) -> formatter -> 'c -> 'd
```

Comment:
Is it really worth having `'b` and `'d` different from `unit` ? 

Suppose one day the `Stdlib` introduces the abbreviation `'a Format.t = Format.formatter -> 'a -> unit` to clarify these long winded formatter signatures. We won't be able to use it here (unless compat is broken).